libbeat/processors/add_host_metadata/add_host_metadata.go

```diff
@@ -48,23 +63,30 @@ type addHostMetadata struct {
 	geoData mapstr.M
 	config  Config
 	logger  *logp.Logger
+	metrics metrics
 }
 
-const (
-	processorName = "add_host_metadata"
-)
-
 // New constructs a new add_host_metadata processor.
 func New(cfg *config.C) (processors.Processor, error) {
 	c := defaultConfig()
 	if err := cfg.Unpack(&c); err != nil {
 		return nil, fmt.Errorf("fail to unpack the %v configuration: %w", processorName, err)
 	}
 
+	// Logging and metrics (each processor instance has a unique ID).
+	var (
+		id     = int(instanceID.Inc())
+		logger = logp.NewLogger(logName).With("instance_id", id)
+		reg    = monitoring.Default.NewRegistry(logName+"."+strconv.Itoa(id), monitoring.DoNotReport)
```

Comment:
Do we actually need a separate instance of this per processor? Do we expect the errors to be unique per processor instance? There's also no way to map the ID here back to the configuration, so it seems a bit arbitrary and is probably simpler if we just have one atomic counter shared by all instances of this processor.

```json
      "processor": {
        "add_host_metadata": {
          "1": {
            "fqdn_lookup_failed": 1
          }
        }
      },
```

99% of the time there is just going to be the one add_host_metadata instance applied to each event collected since that is what is configured by default. If someone configured this to be per input we could end up with a lot of processor instances (think k8s autodiscovery where every pod gets its own input).

Reply:
Fixed via d9d55c7735.